alg/crc32c_sse42.c

```diff
@@ -15,9 +17,15 @@
 void
 CRC32C_Update_SSE42(uint32_t * state, const uint8_t * buf, size_t len)
 {
-	const size_t block_len = len - len % 8;
+	const size_t pre = ((uintptr_t)buf & 7) < len ? (uintptr_t)buf & 7 : len;
+	const size_t remaining = ((len - pre) < len) ? len - pre : 0;
```

Comment:
This is rather ugly.  I'm quite curious to hear the proper way of making sure that we're loading values with proper alignment.